In [1]:
import os

In [2]:
%pwd

'c:\\Users\\Venks\\Desktop\\Project\\MLOPS_Bootcamp\\Deep Learning\\Computer_Vision\\Aerial_Object_Classification_And_Detection\\research'

In [3]:
os.chdir(".. /")

FileNotFoundError: [WinError 2] The system cannot find the file specified: '.. /'

In [4]:
%pwd

'c:\\Users\\Venks\\Desktop\\Project\\MLOPS_Bootcamp\\Deep Learning\\Computer_Vision\\Aerial_Object_Classification_And_Detection\\research'

Testing pytorch: 
- Error encountered as directly library was imported instead of using the cudnn- enabled activation 
- Wrong Python version installed earlier, changed from 3.8 ->3.10
- Video: https://www.youtube.com/watch?v=nATRPPZ5dGE&t=282s
- Official link to download pytorch: https://pytorch.org/get-started/locally/

In [6]:
!nvidia-smi

Tue Nov 18 10:16:23 2025       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 560.81                 Driver Version: 560.81         CUDA Version: 12.6     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                  Driver-Model | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 3050 ...  WDDM  |   00000000:01:00.0 Off |                  N/A |
| N/A   68C    P0              9W /   35W |       0MiB /   4096MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+-----

In [4]:
import torch
torch.cuda.is_available()

True

In [5]:

print("Number of GPU: ", torch.cuda.device_count())
print("GPU Name: ", torch.cuda.get_device_name())




Number of GPU:  1
GPU Name:  NVIDIA GeForce RTX 3050 Laptop GPU


In [7]:
!pip list

Package                   Version
------------------------- ------------
altair                    5.5.0
asttokens                 3.0.1
attrs                     25.4.0
beautifulsoup4            4.14.2
blinker                   1.9.0
boto3                     1.40.75
botocore                  1.40.75
cachetools                6.2.2
certifi                   2025.11.12
charset-normalizer        3.4.4
click                     8.3.1
colorama                  0.4.6
comm                      0.2.3
contourpy                 1.3.2
cycler                    0.12.1
debugpy                   1.8.17
decorator                 5.2.1
exceptiongroup            1.3.0
executing                 2.2.1
filelock                  3.20.0
fonttools                 4.60.1
fsspec                    2025.10.0
gdown                     5.2.0
gitdb                     4.0.12
GitPython                 3.1.45
idna                      3.11
ipykernel                 7.1.0
ipython                   8.37.0
jedi      

In [ ]:
#!pip uninstall torch torchvision

^C


Pytorch is successfully set up with cuda version 12.6 if tensorflow gpu has to be set up then have to change the installed cuda version 12.6 -> cudatoolkit=11.2 cudnn=8.1.0